### ISSUE TITLE
[$250] LHN - Member WS chat title on LHN updates only after revisiting the chat

### ISSUE THREAD

**Issue by vincdargento** [Expensify/App#53473]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.70-2
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** testpayment935+737@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Navigate to https://staging.new.expensify.com/
2. Create WS and add member
3. Without navigating to members WS chat, notice members chat title from WS remain the WS name rather than Member name

## Expected Result:
Each member have WS chat on LHN to submit expense with their name as the Title without visiting the chat

## Actual Result:
The member WS chat name persist the WS name until navigating to the chat

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/a18e6b3b-fde7-4652-8502-ad85f29f38a2

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021864470970450528640</li>
        <li>Upwork Job ID: 1864470970450528640</li>
        <li>Last Price Increase: 2024-12-05</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @hoangzinh</details>

**Comment by Tony-MK** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.


LHN - Member WS chat title on LHN updates only after revisiting the chat

### What is the root cause of that problem?

The [`getPolicyExpenseChatName`](https://github.com/Expensify/App/blob/acf418ced84cc7066476d8b7cce7fd52b8397f01/src/libs/ReportUtils.ts#L2891) returns the policy name because `report?.isOwnPolicyExpenseChat` is `true`.

https://github.com/Expensify/App/blob/acf418ced84cc7066476d8b7cce7fd52b8397f01/src/libs/ReportUtils.ts#L2891-L2901

This is because `AddMembersToWorkspace` returns `report?.isOwnPolicyExpenseChat` as `true` in its response. 

Only after the user clicks on the report and `openReport` is called is when the `report?.isOwnPolicyExpenseChat` set as `false`.

### What changes do you think we should make in order to solve the problem?


From the backend, edit the `AddMembersToWorkspace` response to return `report?.isOwnPolicyExpenseChat` as `false` instead of `false` for these types of reports.

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

From the frontend, change to the condition below to check if the `report.ownerAccountID` is equal to the `currentUserAccountID`.

https://github.com/Expensify/App/blob/acf418ced84cc7066476d8b7cce7fd52b8397f01/src/libs/ReportUtils.ts#L2899-L2900

```
if (report?.isOwnPolicyExpenseChat && report?.ownerAccountID === currentUserAccountID) {
    return getPolicyName(report, false, policy); 
```

**Comment by Tony-MK** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.


LHN - Member WS chat title on LHN updates only after revisiting the chat

### What is the root cause of that problem?

The [`getPolicyExpenseChatName`](https://github.com/Expensify/App/blob/acf418ced84cc7066476d8b7cce7fd52b8397f01/src/libs/ReportUtils.ts#L2891) returns the policy name because `report?.isOwnPolicyExpenseChat` is `true`.

https://github.com/Expensify/App/blob/acf418ced84cc7066476d8b7cce7fd52b8397f01/src/libs/ReportUtils.ts#L2891-L2901

This is because `AddMembersToWorkspace` returns `report?.isOwnPolicyExpenseChat` as `true` in its response. 

Only after the user clicks on the report and `openReport` is called is when the `report?.isOwnPolicyExpenseChat` set as `false`.

### What changes do you think we should make in order to solve the problem?


From the backend, edit the `AddMembersToWorkspace` response to return `report?.isOwnPolicyExpenseChat` as `false` instead of `false` for these types of reports.

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


N/A

### What alternative solutions did you explore? (Optional)

From the frontend, change to the condition below to check if the `report.ownerAccountID` is equal to the `currentUserAccountID`.

https://github.com/Expensify/App/blob/acf418ced84cc7066476d8b7cce7fd52b8397f01/src/libs/ReportUtils.ts#L2899-L2900

```
if (report?.isOwnPolicyExpenseChat && report?.ownerAccountID === currentUserAccountID) {
    return getPolicyName(report, false, policy); 
```

**Comment by RachCHopkins** [comment]
Took me a while to repro - maybe a network/wifi speed issue that makes it obvious. I was able to get a fleeting glance of the behaviour on Safari only.

**Comment by Tony-MK** [comment]
I believe this issue would be best solved internally.

**Comment by RachCHopkins** [comment]
On reflection, with this being Low priority, I think Monthly is more realistic here.
